### PR TITLE
Make globalstep optional

### PR DESCRIPTION
**Pull request by MoNTE48**
1) Now you can sit down and get up by clicking on a chair.
2) So globalstep becomes optional, which may be useful for the server.
3) The screenshot was compressed without loss of quality.
4) A few minor fixes

### CODE REVIEW

**Comment by Thomas--S** [comment]
Thank you for your contribution, this looks very promising.

I will try to review and test this as soon as possible.

File: init.lua [hunk 1]
```diff
@@ -2,48 +2,59 @@ ts_furniture = {}
...
-		default.player_attached[name] = true
-		default.player_set_animation(player, "sit", 30)
+	-- The player will stand at the beginning of the movement
+	if ts_furniture.enable_globalstep and not minetest.get_modpath("cozy") then
+		minetest.register_globalstep(function(dtime)
+			local players = minetest.get_connected_players()
+			for i = 1, #players do
+				local name = players[i]:get_player_name()
+				if default.player_attached[name] and not players[i]:get_attach() and
+						(players[i]:get_player_control().up    == true or
```
Comment (MoNTE48) [inline init.lua]: ToDo: simplify it

**Comment by MoNTE48** [comment]
Simplified the code a bit more. Tell me if something is wrong.
Very cool mod, by the way. I felt I had to do backport...

File: settingtypes.txt [hunk 2]
```diff
@@ -1 +1,2 @@
-ts_furniture.enable_sitting (Enable sitting) bool true
\ No newline at end of file
+ts_furniture.enable_sitting (Enable sitting) bool true
+ts_furniture.enable_globalstep (Enable enable_globalstep) bool true
```
Comment (Thomas--S) [inline settingtypes.txt]:
```suggestion
ts_furniture.enable_globalstep (Enable globalstep) bool true
```

**Review by Thomas--S — APPROVED** [review]
Tested and seems to work fine.

Code also looks good as far as I can see.

Thanks for your contribution. I greatly appreciated it.

Is this PR ready for merge or do you plan additional changes?

**Comment by MoNTE48** [comment]
@Thomas--S, there was something else, but I will leave it for the future PR.
Changed what you requested. I was glad to help!

**Pull request merged**